experimental/unitsconversion/src/converter.rs

```diff
@@ -0,0 +1,256 @@
+        ) -> Result<(), ConversionError> {
+            for item in units {
+                let items_from_item = factory
+                    .payload
+                    .convert_infos
+                    .get(item.unit_id as usize)
+                    .ok_or(ConversionError::DataNotFoundError)?;
+
+                insert_base_units(items_from_item.basic_units(), item.power as i16, sign, map);
+            }
+
+            Ok(())
+        }
+
+        /// Inserting the basic units into the map.
+        /// NOTE:   
+        ///     The base units should be multiplied by the original power.
+        ///     For example, `square-foot` , the base unit is `meter` with power 1.
+        ///     Thus, the inserted power should be `1 * 2 = 2`.
+        fn insert_base_units(
+            basic_units: &ZeroSlice<MeasureUnitItem>,
+            original_power: i16,
+            sign: i16,
+            map: &mut LiteMap<u16, PowersInfo>,
+        ) {
+            for item in basic_units.iter() {
+                let item_power = (item.power as i16) * original_power;
+                let signed_item_power = item_power * sign;
+                if let Some(powers) = map.get_mut(&item.unit_id) {
+                    powers.diffs += signed_item_power;
+                    powers.sums += item_power;
+                } else {
+                    map.insert(
+                        item.unit_id,
+                        PowersInfo {
+                            diffs: (signed_item_power),
+                            sums: (item_power),
+                        },
+                    );
+                }
+            }
+        }
+
+        let unit1 = &unit1.contained_units;
+        let unit2 = &unit2.contained_units;
+
+        let mut map = LiteMap::new();
+        let first_insert_result = insert_non_basic_units(self, unit1, 1, &mut map);
+        let second_insert_result = insert_non_basic_units(self, unit2, -1, &mut map);
+
+        debug_assert!(first_insert_result.is_ok());
+        debug_assert!(second_insert_result.is_ok());
+
+        if first_insert_result.is_err() || second_insert_result.is_err() {
+            return None;
+        }
+
+        let (power_sums_are_zero, power_diffs_are_zero) =
+            map.iter_values()
+                .fold((true, true), |(sums, diffs), determine_convertibility| {
+                    (
+                        sums && determine_convertibility.sums == 0,
+                        diffs && determine_convertibility.diffs == 0,
+                    )
+                });
+
+        if power_diffs_are_zero {
+            Some(false)
+        } else if power_sums_are_zero {
+            Some(true)
+        } else {
+            None
+        }
+    }
+
+    fn apply_si_prefix(si_prefix: &SiPrefix, ratio: &mut Ratio<BigInt>) {
+        match si_prefix.base {
+            Base::Decimal => {
+                *ratio *= Ratio::<BigInt>::from_integer(10.into()).pow(si_prefix.power as i32);
```

Comment:
You don't need to create a `Ratio`, you can multiply directly with a `BigInt`:

```suggestion
                *ratio *= BigInt::from(10).pow(si_prefix.power as i32);
```

Reply:
`BigInt::pow` only accepts `u32` .